### PR TITLE
fix: unify delete logic and fix cascade deletion for flows

### PR DESCRIPTION
**Pull request by dhlidongming**
This PR fixes two bugs related to the flow deletion process:
1. The previous method in `delete_multiple_flows` could leave behind undeleted entries in `TransactionTable` and `VertexBuildTable` due to the default limit parameter of 1000 in `get_transactions_by_flow_id` and `get_vertex_builds_by_flow_id`. This PR unifies the logic between `delete_flow` and `delete_multiple_flows` by using the `cascade_delete_flow` function:
https://github.com/langflow-ai/langflow/blob/0118e98b5c432ba0690c8feffedeb6808e91d4c9/src/backend/base/langflow/api/v1/flows.py#L319-L335
ensuring complete removal of all related records.
2. The `cascade_delete_flow` function has been updated to include deletion of `MessageTable` entries. This prevents foreign key constraints from resulting in `NULL` values in the `flow_id` column after a flow is deleted, ensuring the integrity and cleanliness of data across the database.

### CODE REVIEW

File: src/backend/base/langflow/api/utils.py [hunk 1]
```diff
@@ -283,6 +283,7 @@ async def cascade_delete_flow(session: AsyncSession, flow_id: uuid.UUID) -> None
     try:
         await session.exec(delete(TransactionTable).where(TransactionTable.flow_id == flow_id))
         await session.exec(delete(VertexBuildTable).where(VertexBuildTable.flow_id == flow_id))
+        await session.exec(delete(MessageTable).where(MessageTable.flow_id == flow_id))
```
Comment (ogabrielluiz) [inline src/backend/base/langflow/api/utils.py:286]: I'm not sure this is a good idea. Even though we tie the message to a flow and if we set it to null it won't appear in any playground anymore, the session id, which is what we use to search for messages might still be used. 

Could you comment this and add a TODO here for us to verify this behavior later?

**Review by ogabrielluiz — CHANGES_REQUESTED** [review]
Hey, @dhlidongming 

thanks for the PR

File: src/backend/base/langflow/api/utils.py (hunk 1 above)
Comment (dhlidongming) [reply src/backend/base/langflow/api/utils.py]: Thank you for the feedback. I have added a comment and a TODO. Please review the changes.

**Review by ogabrielluiz — APPROVED** [review]
Thank you!

LGTM

**Comment by jeevic** [comment]
can't delete flow  because  message table foreign key constraint

**Pull request merged**